tests/Codec/LazyBSONCodecLibraryTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace MongoDB\Tests\Codec;
+
+use Generator;
+use MongoDB\BSON\Document;
+use MongoDB\BSON\PackedArray;
+use MongoDB\Codec\LazyBSONCodecLibrary;
+use MongoDB\Model\LazyBSONArray;
+use MongoDB\Model\LazyBSONDocument;
+use MongoDB\Tests\TestCase;
+
+class LazyBSONCodecLibraryTest extends TestCase
+{
+    public static function provideDecodedData(): Generator
+    {
+        $array = [
+            'bar',
+            ['foo' => 'bar'],
```

Comment:
Did you mean for this to be cast to an object or is it intentionally an associative array?

Reply:
@alcaeus: Outstanding question.

Reply:
Sorry for missing this. This was an error, I've changed it to an object.